### PR TITLE
Move copycat dependency to catalyst

### PR DESCRIPTION
**Pull request by LuQQiu**
Remove the copycat dependency and change it to a much smaller catalyst dependency.
Catalyst dependency is needed for Grpc messaging serialization and deserialization.

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 1]
```diff
@@ -1511,15 +1511,6 @@ public String toString() {
           .setDefaultValue(19200)
           .setScope(Scope.ALL)
           .build();
-  public static final PropertyKey MASTER_EMBEDDED_JOURNAL_STORAGE_LEVEL =
```
Comment (bf8086) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:1514]: Is there a way to warn user of the deprecation of this property?

Comment (LuQQiu) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java:1514]: Oh yeah, i wll leave the property key there, i remember it's bunch processed during release

Comment (LuQQiu) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java:1514]: Will add a @deprecated to the key

**Comment by LuQQiu** [comment]
alluxio.server.ft.journal.raft.EmbeddedJournalIntegrationTest
23:25:07.136 [ERROR] alluxio.server.ft.journal.raft.EmbeddedJournalIntegrationTest.restartStress  Time elapsed: 600.023 s  <<< ERROR!
org.junit.runners.model.TestTimedOutException: test timed out after 600000 milliseconds
	at alluxio.server.ft.journal.raft.EmbeddedJournalIntegrationTest.after(EmbeddedJournalIntegrationTest.java:72)

23:25:07.137 [ERROR] alluxio.server.ft.journal.raft.EmbeddedJournalIntegrationTest.restartStress  Time elapsed: 120.146 s  <<< ERROR!
java.util.concurrent.TimeoutException: Timed out waiting for 11 successes options: WaitForOptions{interval=20, timeout=120000} last value: false
	at alluxio.server.ft.journal.raft.EmbeddedJournalIntegrationTest.restartStress(EmbeddedJournalIntegrationTest.java:337)

**Comment by LuQQiu** [comment]
jenkins, test this please

**Comment by LuQQiu** [comment]
jenkins, test this please

**Review by bf8086 — APPROVED** [review]
LGTM.

File: pom.xml [hunk 2]
```diff
@@ -127,7 +127,7 @@
     <apache.curator.version>4.2.0</apache.curator.version>
     <aws.amazonaws.version>1.11.215</aws.amazonaws.version>
     <build.path>build</build.path>
-    <copycat.version>1.2.15</copycat.version>
+    <catalyst.version>1.2.1</catalyst.version>
```
Comment (gpang) [inline pom.xml:130]: was it intentional to switch from version 1.2.15?

**Review by gpang — COMMENTED** [review]
@LuQQiu I left a minor question

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

File: pom.xml (hunk 2 above)
Comment (LuQQiu) [reply pom.xml:130]: oh i didn't see this one
for catalyst the lastest version is 1.2.1
copycat 1.2.15 uses catalyst 1.2.1

**Pull request merged**